src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -957,91 +954,121 @@ private String getBody(ScanResults.XIssue issue, ScanRequest request, String fil
-                        if (request.getBugTracker().getOpenStatus().contains(fpIssue.getStatus().getName())) { //If the status is of open state, close it
-                            /*Close the issue*/
-                            log.info("Closing issue with key {}", fpIssue.getKey());
-                            this.transitionCloseIssue(fpIssue.getKey(), request.getBugTracker().getCloseTransition(), request.getBugTracker(), true);
-                            closedIssues.add(fpIssue.getKey());
-                        }
+                        fpIssue = checkForFalsePositiveIssuesInList(request, xIssue, currentIssue, issue);
+                        closeIssueInCaseOfIssueIsInOpenState(request, closedIssues, fpIssue);
                     }/*Ignore any with label indicating false positive*/
-                    else if (!i.getLabels().contains(jiraProperties.getFalsePositiveLabel())) {
-                        log.debug("Issue still exists.  Updating issue with key {}", xIssue.getKey());
-                        Issue updatedIssue = this.updateIssue(i.getKey(), currentIssue, request);
-                        if (updatedIssue != null) {
-                            log.debug("Update completed for issue #{}", updatedIssue.getKey());
-                            updatedIssues.add(updatedIssue.getKey());
-                            if (jiraProperties.isUpdateComment() && !ScanUtils.empty(jiraProperties.getUpdateCommentValue())) {
-                                addCommentToBug(i.getKey(), jiraProperties.getUpdateCommentValue());
-                            }
-                        }
+                    else if (!issue.getLabels().contains(jiraProperties.getFalsePositiveLabel())) {
+                        updateIssue(request, updatedIssues, xIssue, currentIssue, issue);
                     } else {
                         log.info("Skipping issue marked as false-positive or has False Positive state with key {}", xIssue.getKey());
                     }
                 } else {
                     /*Create the new issue*/
-                    if(!currentIssue.isAllFalsePositive()) {
-                        if (!jiraProperties.isChild() || (!parentCheck(xIssue.getKey(), issuesParent) && !grandparentCheck(xIssue.getKey(), issuesGrandParent))) {
-                            if (jiraProperties.isChild()) {
-                                log.info("Issue not found in parent creating issue for child");
-                            }
-
-                            log.debug("Creating new issue with key {}", xIssue.getKey());
-                            String newIssue = this.createIssue(currentIssue, request);
-                            newIssues.add(newIssue);
-                            log.info("New issue created. #{}", newIssue);
+                    if (!currentIssue.isAllFalsePositive() && (!jiraProperties.isChild() || (!parentCheck(xIssue.getKey(), issuesParent) && !grandparentCheck(xIssue.getKey(), issuesGrandParent)))) {
```

Comment:
Need to improve readability of this if statement.
You can define booleans as local variables with explanatory names and then check for them inside if
if not, at least split the line to several lines on screen